assets/scripts/components/table-of-contents.js

```diff
@@ -240,5 +246,24 @@ window.addEventListener('scroll', () => {
     onScroll();
 });
 
-DOMReady(handleAPIPage);
+// Expose the necessary functions to cdocs
+window.cdocsHooks = {};
+
+const phases = ['beforeReveal', 'afterReveal', 'afterRerender'];
+phases.forEach((phase) => {
+    window.cdocsHooks[phase] = [];
```

Comment:
This line is slightly out of place in that the cdocs integration should probably be the one to establish the empty hooks array. But I wasn't sure what order all of the site's scripts would run in by the time cdocs was merged, and the hooks are not used anywhere but in this file. If it's preferred that the empty array declaration is in the cdocs integration, I can put it there, it just is a little more convoluted for our needs even if it makes more architectural sense.

Reply:
Update, after talking through this with a few people, I've changed my mind, we should fix it -- the team is hoping to tweak our tabs in a way that would also require hooks, so we should just use a good pattern now. I'll update the integration and make a PR against this branch tomorrow, we can just do something like `clientFiltersManager.addHook("afterReveal", buildTOCMap)` here to add each hook.

I think it's still fine to review the rest of the PR while I do that, or feel free to move it back to draft if you'd rather just get it reviewed all at once.

Reply:
[PR in the packages repo to fix this](https://github.com/DataDog/corp-node-packages/pull/103), once that's approved I can update this PR as well.